### PR TITLE
51 tiling

### PR DESCRIPTION
**Pull request by dalessioluca**
something is wrong...

### CODE REVIEW

File: MODULES/non_max_suppression.py [hunk 1]
```diff
@@ -109,32 +106,26 @@ def compute_mask_and_index(prob: torch.Tensor,
...
 
         if topk_only:
             # If nms_mask = 1 then this is equivalent to do topk only
-            chosen_nms_mask = torch.ones_like(noisy_score)
+            chosen_nms_mask = torch.ones_like(prob)
         else:
-            possible = (noisy_score > score_threshold).float()
-            chosen_nms_mask: torch.Tensor = NonMaxSuppression.perform_nms_selection(mask_overlap=binarized_overlap_measure,
-                                                                                    score=noisy_score,
+            possible = (prob > 0.0).float()
```
Comment (dalessioluca) [inline MODULES/non_max_suppression.py:116]: At the beginning all prob should be possible so that they have a chance of receiving gradients.
Change (prob > 0.0).float() -> (prob > -1.0).float()

File: MODULES/vae_parts.py [hunk 2]
```diff
@@ -238,9 +242,38 @@ def forward(self, imgs_in: torch.Tensor,
...
+                    n_to_keep = n_element - n_to_drop
+                    index_to_keep = torch.topk(weights, n_to_keep, dim=-1, largest=True, sorted=True)[1]
+                    batch_index = torch.arange(batch_size,
+                                               dtype=index_to_keep.dtype,
+                                               device=index_to_keep.device).view(-1, 1)
+                    mask_dropout = torch.zeros_like(weights)
+                    mask_dropout[batch_index, index_to_keep] = 1.0
+                mask_dropout = mask_dropout.view_as(p_map)
+            else:
+                mask_dropout = torch.ones_like(p_map)
```
Comment (dalessioluca) [inline MODULES/vae_parts.py:271]: dropout does not seem to help. Remove it?

File: MODULES/vae_model.py [hunk 3]
```diff
@@ -215,17 +204,26 @@ def NLL_MSE(self, output: torch.tensor, target: torch.tensor, sigma: torch.tenso
...
+                        reg: RegMiniBatch,
+                        dropout_prob: float) -> MetricMiniBatch:
 
         # Preparation
+        batch_size, ch, w, h = imgs_in.shape
         n_obj_counts = (inference.prob > 0.5).float().sum(-2).detach()  # sum over boxes dimension
         p_times_area_map = inference.p_map * inference.area_map
         mixing_k = inference.big_mask * inference.prob[..., None, None, None]
         mixing_fg = torch.sum(mixing_k, dim=-5)
         mixing_bg = 1.0 - mixing_fg
```
Comment (dalessioluca) [inline MODULES/vae_model.py:216]: use torch.ones_like
